tool/sdg/src/main/java/org/openscience/cdk/layout/StructureDiagramGenerator.java

```diff
@@ -326,19 +499,20 @@ public void generateCoordinates(Vector2d firstBondVector) throws CDKException {
      *
      * @param firstBondVector the vector of the first bond to lay out
      * @param isConnected     the 'molecule' attribute is guaranteed to be connected (we have checked)
-     * @param isSubLayout     the 'molecule' attribute is guaranteed to be connected (we have checked)
+     * @param isSubLayout     the 'molecule' is being laid out as part of a large collection of fragments
      * @throws CDKException problem occurred during layout
      */
     private void generateCoordinates(Vector2d firstBondVector, boolean isConnected, boolean isSubLayout) throws CDKException {
 
-        int safetyCounter = 0;
-        /*
-         * if molecule contains only one Atom, don't fail, simply set
-         * coordinates to simplest: 0,0. See bug #780545
-         */
+        final int numAtoms = molecule.getAtomCount();
+        final int numBonds = molecule.getBondCount();
+        this.firstBondVector = firstBondVector;
+
+        // if molecule contains only one Atom, don't fail, simply set
+        // coordinates to simplest: 0,0. See bug #780545
         logger.debug("Entry point of generateCoordinates()");
-        logger.debug("We have a molecules with " + molecule.getAtomCount() + " atoms.");
-        if (molecule.getAtomCount() == 1) {
+        logger.debug("We have a molecules with " + numAtoms + " atoms.");
```

Comment:
As earlier: 

logger.debug("We have a molecules with ", numAtoms, " atoms.");

Saves two String created to be garbaged later...